src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/runtime/ObjectSynchronizer.java

```diff
@@ -93,46 +92,34 @@ public static Iterator objectMonitorIterator() {
 
   private static class ObjectMonitorIterator implements Iterator {
 
-    // JVMTI raw monitors are not pointed by gBlockList
-    // and are not included by this Iterator. May add them later.
-
     ObjectMonitorIterator() {
-      blockAddr = gBlockList;
-      index = blockSize - 1;
-      block = new ObjectMonitor(blockAddr);
+      headAddr = inUseList;
+      mon = new ObjectMonitor(headAddr);
```

Comment:
Why do we need the `headAddr` private field? Why can't we just do:

`mon = new ObjectMonitor(inUseList);`

?